java/vector/src/main/java/org/apache/arrow/vector/VectorSchemaRoot.java

```diff
@@ -312,5 +315,64 @@ public VectorSchemaRoot slice(int index, int length) {
     return new VectorSchemaRoot(sliceVectors);
   }
 
+  /**
+   * Determine if two record batches are exactly equal.
+   */
+  public boolean equals(VectorSchemaRoot other) {
+    if (other == null) {
+      return false;
+    }
+
+    if (!this.schema.equals(other.schema)) {
+      return false;
+    }
+
+    if (this.rowCount != other.rowCount) {
+      return false;
+    }
+
+    for (int i = 0; i < fieldVectors.size(); i++) {
+      FieldVector vector = fieldVectors.get(i);
+      FieldVector otherVector = other.fieldVectors.get(i);
+      if (!VectorEqualsVisitor.vectorEquals(vector, otherVector)) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Determine if two record batches are approximately equal.
+   */
+  public boolean approxEquals(VectorSchemaRoot other) {
```

Comment:
Should we have an override that allows us to pass the epsilon values to the ApproxEqualsVisitor, and/or to provide an already created ApproxEqualsVisitor?

Reply:
Thanks a lot for your comments, I add an override allows to pass diffFunction(which also contains tolerances) to calculate difference between float/double values.